charts/kube-master/charts/etcd/templates/deployment.yaml

```diff
@@ -42,33 +76,60 @@ spec:
               valueFrom:
                 fieldRef:
                   fieldPath: status.podIP
-          command: [ etcd ]
-          args:
-            - --name=kubernikus
-            - --data-dir=/var/lib/etcd
-            - --advertise-client-urls=http://$(ETCD_IP):2379
-            - --initial-advertise-peer-urls=http://$(ETCD_IP):2380
-            - --initial-cluster=kubernikus=http://$(ETCD_IP):2380
-            - --listen-client-urls=http://0.0.0.0:2379
-            - --listen-peer-urls=http://$(ETCD_IP):2380
-            {{- if false }}
-            - --trusted-ca-file=/etc/etcd/certs/etcd-clients/ca.pem
-            - --cert-file=/etc/etcd/certs/etcd-clients/etcd-server.pem
-            - --key-file=/etc/etcd/certs/etcd-clients/etcd-server-key.pem
-            - --peer-ca-file=/etc/etcd/certs/etcd-peers/ca.pem
-            - --peer-cert-file=/etc/etcd/certs/etcd-peers/etcd-0.pem
-            - --peer-key-file=/etc/etcd/certs/etcd-peers/etcd-0-key.pem
-            {{- end }}
+          command:
+            - /bootstrap/bootstrap.sh
           volumeMounts:
-            {{- if false }}
-            - mountPath: /etc/etcd/certs/etcd-clients
-              name: etcd-clients
-              readOnly: true
-            - mountPath: /etc/etcd/certs/etcd-peers
-              name: etcd-peers
-              readOnly: true
-            {{- end }}
             - mountPath: /var/lib/etcd
               name: data
+            - mountPath: /bootstrap
+              name: {{ include "fullname" . }}-bootstrap
           resources:
+        - name: backup
+          command:
+            - etcdbrctl
+            - server
+            - --schedule=*/5 * * * *
+            - --max-backups=10
+            - --data-dir=/var/lib/etcd
+            - --insecure-transport=true
+            - --storage-provider=Swift
+            - --delta-snapshot-period-seconds=10
+            - --garbage-collection-period-seconds=60
+          image: eu.gcr.io/gardener-project/gardener/etcdbrctl:0.3.0
+          imagePullPolicy: Always
+          ports:
+            - containerPort: 8080
+              name: server
+              protocol: TCP
+          env:
+            - name: STORAGE_CONTAINER
+              value: etcd-backup-{{ .Values.klustername }}-{{ uuidv4 }}
```

Comment:
This will get pretty long. Did you check this actually works? 😛 

Also, when does `uuidv4` change? Couldn't it happen that we change something in this Helm chart and we get a new UID, losing the old backup? Not support critical but potentially a source of many garbage backups.

Reply:
This will generate something like "etcd-backup-mykluster-4f1d1d5a-d845-4bfb-940b-be6b444497a1". We need a kluster-unique value here to identify the correct backup set. This might become a problem when upgrading helm chart. Any ideas?

Reply:
`etcd-backup-mykluster-4f1d1d5a-d845-4bfb-940b` is already unique. At least as long as the kluster isn't deleted and recreated with the same name.